### PR TITLE
Fix open TODO in music_cont_in_proxy

### LINKED ISSUE #1390: Open TODO in `music_cont_in_proxy.cpp`
During review of #1245 we encountered an open TODO in `music_cont_in_proxy.cpp`. This should most likely be fixed or turned into a helpful comment.

Unfortunately the history stops at the move from svn to git. Maybe @mdjurfeldt @jougs remember what this was about?

### PR DESCRIPTION
**Pull request by jougs**
Fixes #1390

### CODE REVIEW

File: models/music_cont_in_proxy.cpp [hunk 1]
```diff
@@ -167,11 +168,11 @@ nest::music_cont_in_proxy::get_status( DictionaryDatum& d ) const
 void
 nest::music_cont_in_proxy::set_status( const DictionaryDatum& d )
 {
-  Parameters_ ptmp = P_; // temporary copy in case of errors
-  ptmp.set( d, S_ );     // throws if BadProperty
+  Parameters_ ptmp = P_;    // temporary copy in case of errors
```
Comment (med-ayssar) [inline models/music_cont_in_proxy.cpp:171]: Why do we need a temp attribute here? If something went wrong in the `set` function, then an exception will be thrown, and we end the execution.

Comment (jougs) [reply models/music_cont_in_proxy.cpp:171]: But then we might already have set half of the new parameters on P_, and the user has now way to know which worked and which did not.

This same pattern is used in every model:
1. create a temporary copy
1. check if setting on the copy works
1. if yes, swap the updated copy with the original parameters.

File: models/music_cont_in_proxy.cpp [hunk 2]
```diff
@@ -65,11 +65,12 @@ nest::music_cont_in_proxy::Parameters_::get( DictionaryDatum& d ) const
 }
 
 void
-nest::music_cont_in_proxy::Parameters_::set( const DictionaryDatum& d, State_& s )
+nest::music_cont_in_proxy::Parameters_::set( const DictionaryDatum& d, State_& s, Node* node )
```
Comment (heplesser) [inline models/music_cont_in_proxy.cpp:68]:
```suggestion
nest::music_cont_in_proxy::Parameters_::set( const DictionaryDatum& d, State_& s, const Node& node )
```

File: models/music_cont_in_proxy.cpp [hunk 3]
```diff
@@ -65,11 +65,12 @@ nest::music_cont_in_proxy::Parameters_::get( DictionaryDatum& d ) const
...
 {
-  // TODO: This is not possible, as P_ does not know about get_name()
-  //  if(d->known(names::port_name) and s.published_)
-  //    throw MUSICPortAlreadyPublished(get_name(), P_.port_name_);
+  if( d->known(names::port_name) and s.published_ )
+  {
+    throw MUSICPortAlreadyPublished(node->get_name(), port_name_);
```
Comment (heplesser) [inline models/music_cont_in_proxy.cpp:72]:
```suggestion
    throw MUSICPortAlreadyPublished(node.get_name(), port_name_);
```

File: models/music_cont_in_proxy.cpp [hunk 4]
```diff
@@ -167,11 +168,11 @@ nest::music_cont_in_proxy::get_status( DictionaryDatum& d ) const
 void
 nest::music_cont_in_proxy::set_status( const DictionaryDatum& d )
 {
-  Parameters_ ptmp = P_; // temporary copy in case of errors
-  ptmp.set( d, S_ );     // throws if BadProperty
+  Parameters_ ptmp = P_;    // temporary copy in case of errors
+  ptmp.set( d, S_, this );  // throws if BadProperty
```
Comment (heplesser) [inline models/music_cont_in_proxy.cpp:172]:
```suggestion
  ptmp.set( d, S_, *this );  // throws if BadProperty
```

**Review by heplesser — CHANGES_REQUESTED** [review]
@jougs Thanks! I added suggestions to use `const&` instead of pointer.

**Comment by heplesser** [comment]
OK! And now the FULL run fails again ...

**Review by heplesser — APPROVED** [review]

**Review by med-ayssar — APPROVED** [review]

File: models/music_cont_in_proxy.cpp (hunk 1 above)
Comment (jougs) [reply models/music_cont_in_proxy.cpp]: @med-ayssar: can you please resolve this conversation if you are happy with my answer? We can only merge once all conversations are resolved.

**Pull request merged**